# HG changeset patch
# User rk
# Date 1285926121 -7200
# Node ID 7489a3b6fb109cc9cc3b56e1660de1ba28e9a650
# Parent  7f70824841b6bbbc0dd8a4351b420acce53eeacb
explicitly set the focus to the dataview ctrl window when any mouse button is pressed not just the left one

diff -r 7f70824841b6 -r 7489a3b6fb10 src/generic/datavgen.cpp
--- a/src/generic/datavgen.cpp	Fri Oct 01 11:38:48 2010 +0200
+++ b/src/generic/datavgen.cpp	Fri Oct 01 11:42:01 2010 +0200
@@ -3443,6 +3443,10 @@ void wxDataViewMainWindow::OnMouse( wxMo
         return;
     }
 
+    // set the focus to ourself if any of the mouse buttons are pressed
+    if(event.ButtonDown() && !HasFocus())
+        SetFocus();
+
     int x = event.GetX();
     int y = event.GetY();
     m_owner->CalcUnscrolledPosition( x, y, &x, &y );
@@ -3718,8 +3722,6 @@ void wxDataViewMainWindow::OnMouse( wxMo
     }
     else if ((event.LeftDown() || simulateClick) && !hoverOverExpander)
     {
-        SetFocus();
-
         m_lineBeforeLastClicked = m_lineLastClicked;
         m_lineLastClicked = current;
 
